upi/openstack/network.yaml

```
@@ -32,24 +34,18 @@
       ipv6_ra_mode: "{{ os_subnet6_router_advertisements_mode }}"
     when: os_subnet6 is defined
 
+  - name: 'Prepare list of subnets for router interfaces'
+    set_fact:
+      router_interfaces: " {{ router_interfaces + [ os_subnet6 ] }} "
+    when: os_subnet6 is defined
+
   - name: 'Create external router'
     openstack.cloud.router:
       name: "{{ os_router }}"
       network: "{{ os_external_network }}"
-      interfaces:
-      - "{{ os_subnet }}"
+      interfaces: "{{ router_interfaces }}"
```

The job [currently fails with](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_installer/8087/pull-ci-openshift-installer-master-e2e-openstack-dualstack-upi/1765189720091922432/artifacts/e2e-openstack-dualstack-upi/upi-install-openstack/build-log.txt):
```
TASK [Create external router] **************************************************
fatal: [localhost]: FAILED! => {"changed": false, "extra_data": {"data": null, "details": null, "response": "None"}, "msg": "No Subnet found for  ['ocp-subnet-v4'"}
```

I suspect we need to remove the quotes:
```suggestion
      interfaces: {{ router_interfaces }}
```

---

Well, by removing quotas will make ansible-playbook to complain about wrong syntax:
```console
Syntax Error while loading YAML.
  found unacceptable key (unhashable type: 'AnsibleMapping')
....
The offending line appears to be:

      network: "{{ os_external_network }}"
      interfaces: {{ router_interfaces }}
                   ^ here
...
```
no matter if I initialize `router_interfaces` with `"{{ [os_subnet] }}"` or `["{{ os_subnet }}"]`

---

Although I've seen this error before (i.e. taking part of the list as a string `['ocp-subnet-v4'`). Let me dig a little bit more.

---

@gryf another option is to just create a separate task that will create the router with two interfaces